tests/console/yast2_i.pm

```diff
@@ -95,19 +116,37 @@ sub run {
     send_key "alt-a";        # accept
 
     # Whether summary is shown depends on PKGMGR_ACTION_AT_EXIT in /etc/sysconfig/yast2
-    until (get_var('YAST_SW_NO_SUMMARY')) {
-        assert_screen ['yast2-sw-packages-autoselected', 'yast2-sw_automatic-changes', 'yast2-sw_shows_summary'], 60;
-        # automatic changes for manual selections
-        if (match_has_tag('yast2-sw-packages-autoselected') or match_has_tag('yast2-sw_automatic-changes')) {
-            wait_screen_change { send_key 'alt-o' };
-            next;
-        }
-        elsif (match_has_tag('yast2-sw_shows_summary')) {
-            wait_screen_change { send_key 'alt-f' };
-            last;
-        }
+    # Possible actions are:
+    #   close - just finish the package manager
+    #   restart - go back to the package manager, install/remove more packages
+    #   summary - display an installation summary dialog, there user can decide whether to finish or restart
+    assert_screen 'yast2-sw_automatic-changes';
+    wait_screen_change { send_key 'alt-o' };
+    if ($action =~ m/summary/) {
+        assert_screen 'yast2-sw_shows_summary';
+        wait_screen_change { send_key 'alt-f' };
+    } elsif ($action =~ m/restart/) {
+        assert_screen 'yast2-preselected-driver';
+        wait_screen_change { send_key 'alt-c' };
+    } elsif ($action =~ m/close/) {
+        save_screenshot;
+    } else {
+        die "PKGMGR_ACTION_AT_EXIT possible actions (summary|restart|close)!\n";
     }
 
+    #until (get_var('YAST_SW_NO_SUMMARY')) {
```

Comment:
Is this part intended? Or we can even remove this code?

Reply:
No, that part will be completely removed. I have just kept that as a reference :)